tx-pool/src/pool.rs

```diff
@@ -84,6 +87,53 @@ impl TxPool {
         self.total_tx_cycles += cycles;
     }
 
+    /// Check whether tx-pool enable RBF
+    pub fn enable_rbf(&self) -> bool {
+        self.config.min_rbf_rate > self.config.min_fee_rate
+    }
+
+    /// The least required fee rate to allow tx to be replaced
+    pub fn min_replace_fee(&self, tx: &TxEntry) -> Option<Capacity> {
+        if !self.enable_rbf() {
+            return None;
+        }
+        let ids = vec![tx.proposal_short_id()].iter().cloned().collect();
+        self.calculate_min_replace_fee(&ids, tx.size)
+    }
+
+    fn calculate_min_replace_fee(
+        &self,
+        conflicts: &HashSet<ProposalShortId>,
+        size: usize,
+    ) -> Option<Capacity> {
+        let entries = conflicts
+            .iter()
+            .map(|id| {
+                self.get_pool_entry(id)
+                    .expect("conflict Tx should be in pool")
+            })
```

Comment:
To minimize the potential bug impact, I suggest using `filter_map` here instead of `map/expect`